scripts/tslint/preferConstRule.ts

```diff
@@ -0,0 +1,235 @@
+/// <reference path="../../node_modules/tslint/typings/typescriptServices.d.ts" />
+/// <reference path="../../node_modules/tslint/lib/tslint.d.ts" />
+
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING_FACTORY = (identifier: string) => `Identifier '${identifier}' never appears on the LHS of an assignment - use const instead of let for its declaration.`;
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new PreferConstWalker(sourceFile, this.getOptions()));
+    }
+}
+
+function isBindingPattern(node: ts.Node): node is ts.BindingPattern {
+    return !!node && (node.kind === ts.SyntaxKind.ArrayBindingPattern || node.kind === ts.SyntaxKind.ObjectBindingPattern);
+}
+
+function walkUpBindingElementsAndPatterns(node: ts.Node): ts.Node {
+    while (node && (node.kind === ts.SyntaxKind.BindingElement || isBindingPattern(node))) {
+        node = node.parent;
+    }
+
+    return node;
+}
+
+function getCombinedNodeFlags(node: ts.Node): ts.NodeFlags {
+    node = walkUpBindingElementsAndPatterns(node);
+
+    let flags = node.flags;

```

Comment:
That would be an error, good sir. `flags` is mutable and appears on the left hand side of the `|=` operator below.